src/app/comment-period/submit-comment-modal/submit-comment-modal.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, OnInit } from '@angular/core';
+import { CommentPeriodService } from '../../services/comment-period.service';
+import { CommentPeriodComponent } from '../comment-period.component';
+import { RequestOptions } from '@angular/http';
+
+@Component({
+  selector: 'app-submit-comment-modal',
+  templateUrl: './submit-comment-modal.component.html',
+  styleUrls: ['./submit-comment-modal.component.scss']
+})
+export class SubmitCommentModalComponent implements OnInit {
+  files = [];
+  comment;
+  valid: boolean;
+  loading: boolean;
+  error: boolean;
+
+  constructor(private commentPeriodService: CommentPeriodService, private commentPeriodComponent: CommentPeriodComponent) { };
+
+  ngOnInit() {
+    this.comment = {
+      author: '',
+      location: '',
+      visible: false,
+      comment: ''
+    };
+  }
+
+  removeError(event) {
+    event.target.classList.remove('error');
+    event.target.closest('.form-group').classList.remove('has-danger');
+  }
+
+  removeErrors() {
+    const controls = document.querySelectorAll('.form-group');
+    const errors = document.querySelectorAll('.error');
+    for (let i = 0; i < controls.length; i++) {
+      controls[i].classList.remove('has-danger');
+    }
+    for (let n = 0; n < errors.length; n++) {
+      errors[n].classList.remove('error');
+    }
+  }
+
+  displayError(element) {
+    element.closest('.form-group').classList.add('has-danger');
+    element.classList.add('error', 'form-control-danger');
+  }
+
+  validateFields(form) {
+    let size = 0;
+    if (!form.author) {
+      this.displayError(document.getElementById('author'));
+      this.valid = false;
+    }
+    if (!form.location) {
+      this.displayError(document.getElementById('location'));
+      this.valid = false;
+    }
+    if (!form.comment) {
+      this.displayError(document.getElementById('comment'));
+      this.valid = false;
+    }
+    this.files.forEach(file => {
+      size += file.size;
+    });
+    if (size > 5000000) {
+      this.displayError(document.getElementById('file'));
+      return false;
+    }
+    return this.valid;
+  }
+
+  onFileChange(event, form) {
+    const fileInput = <HTMLInputElement>document.getElementById('file');
+    if (event.target.files.length > 0) {
+      const filesList = event.target.files;
+      for (let i = 0; i < filesList.length; i++ ) {
+        if (this.files.length > 0) {
+          const namesArray = this.files.map(file => file.name);
+          if (!namesArray.includes(filesList[i].name)) {
+            this.files.push(filesList[i]);
+          } else {
+            console.log('error stuff here');
```

Comment:
We should update the error message to be descriptive, or remove it altogether if unnecessary.